embulk-jruby-extension/src/main/java/org/jruby/util/ParsedValues.java

```diff
@@ -0,0 +1,406 @@
+        ZONE_OFFSET_MAP.put("w. europe",                3600);
+        ZONE_OFFSET_MAP.put("west asia",               18000);
+        ZONE_OFFSET_MAP.put("west pacific",            36000);
+        ZONE_OFFSET_MAP.put("yakutsk",                 32400);
+    }
+
+    public static int toDiff(String zone)
+    {
+        String z = zone.toLowerCase();
+        int offset = Integer.MIN_VALUE;
+
+        boolean dst = false;
+        if (z.endsWith(" daylight time")) {
+            z = z.substring(0, z.length() - " daylight time".length());
+            dst = true;
+
+        } else if (z.endsWith(" standard time")) {
+            z = z.substring(0, z.length() - " standard time".length());
+            dst = false;
+
+        } else if (z.endsWith(" dst")) {
+            z = z.substring(0, z.length() - " dst".length());
+            dst = true;
+        }
+        if (ZONE_OFFSET_MAP.containsKey(z)) {
+            offset = ZONE_OFFSET_MAP.get(z);
+            if (dst) {
+                offset += 3600;
+            }
+            return offset;
+        }
+
+        if (z.startsWith("gmt") || z.startsWith("utc")) {
+            z = z.substring(3, z.length()); // remove "gmt" or "utc"
+        }
+
+        if (!isSign(z.charAt(0))) {
+            // if z doesn't have "+" or "-", invalid
+            return offset;
+        }
+
+        boolean sign = false;
+        sign = z.charAt(0) == '+';
+        z = z.substring(1, z.length());
+
+        int hour = 0, min = 0, sec = 0;
+        if (z.contains(":")) {
+            String[] splited = z.split(":");
+            if (splited.length == 2) {
+                hour = parseInt(splited[0]);
+                min = parseInt(splited[1]);
+            } else {
+                hour = parseInt(splited[0]);
+                min = parseInt(splited[1]);
+                sec = parseInt(splited[2]);
+            }
+        } else if (z.contains(",") || z.contains(".")) {
+            // TODO min = Rational(fr.to_i, 10**fr.size) * 60
+            String[] splited = z.split("[\\.,]");
+            hour = parseInt(splited[0]);
+            min = (int)(parseInt(splited[1]) * 60 / Math.pow(10, splited[1].length()));
+
+        } else {
+            int len = z.length();
+            if (len % 2 != 0) {
+                if (len >= 1)
+                    hour = parseInt(z.substring(0, 1));
+                if (len >= 3)
+                    min = parseInt(z.substring(1, 3));
+                if (len >= 5)
+                    sec = parseInt(z.substring(3, 5));
+
+            } else {
+                if (len >= 2)
+                    hour = parseInt(z.substring(0, 2));
+                if (len >= 4)
+                    min = parseInt(z.substring(2, 4));
+                if (len >= 6)
+                    sec = parseInt(z.substring(4, 6));
+            }
+        }
+
+        offset = hour * 3600 + min * 60 + sec;
+        return sign ? offset : -offset;
+    }
```

Comment:
You ported date_zone_to_diff in CRuby's ext/date, right? Please provide source reference as a comment.